### PR TITLE
Feat/dockerfile

### LINKED ISSUE #29: Create Docker container

### PR DESCRIPTION
**Pull request by pReya**
closes #29

### CODE REVIEW

**Review by Shorty1o1 — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM rust:1.62.1 as builder
+WORKDIR /usr/src/threematrix
+COPY . .
+RUN cargo fetch --locked
+RUN cargo build --release --frozen --offline
+
+FROM rust:1.62.1
+RUN apt-get update && rm -rf /var/lib/apt/lists/*
```
Comment (EdJoPaTo) [inline Dockerfile:8]: `apt-get update` on its own wont do anything. Only when you install other software the update is required. You can safely omit the update step.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+FROM rust:1.62.1 as builder
+WORKDIR /usr/src/threematrix
+COPY . .
+RUN cargo fetch --locked
+RUN cargo build --release --frozen --offline
+
+FROM rust:1.62.1
```
Comment (EdJoPaTo) [inline Dockerfile:7]: For a smaller container image I would suggest using `docker.io/library/debian:bullseye-slim`. It has everything required to run the single binary (glibc). There is no need to have the rust toolchain on the runtime image.

(Specifying `docker.io/library/` results in a fully specified container image label which is more compatible with other build engines like used in podman)

**Review by EdJoPaTo — COMMENTED** [review]
Some thoughts from my personal container images :innocent:

File: Dockerfile (hunk 1 above)
Comment (pReya) [reply Dockerfile:8]: Thanks, good point!

File: Dockerfile (hunk 2 above)
Comment (pReya) [reply Dockerfile:7]: We tried using `debian:bullseye-slim`. Unfortunately, running our executable failed silently on this image. We were not sure of the exact reason, end especially we did not understand why we got no error for this, but it didn't work. That's why we used the `rust` base image, also for our runtime container. The `rust` container is also based on the official debian images. Maybe we could still try to use `rust:1.62.1-slim` for runtime, to narrow down the reason why the `bullsyeye-slim` image is failing in our use case.

**Pull request merged**